src/pages/index.js

```diff
@@ -4,107 +4,237 @@ import Card from "../components/Card.js";
+  // popup is the popup instance
+  // loadingText is optional for the submit button when the api is being requested
+  popup.renderLoading(true, loadingText); // renders faster when synced before the async begins
+  request()
+    .then(() => {
+      popup.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      popup.renderLoading(false);
+    });
+}
 /* END DECLARATIVE SECTION */
 
-/* PROFILE SECTION */
+/* PROFILE INFO SECTION */
 const userProfile = new UserInfo({
+  // pass as arguments the HTML DOM profile elements
   name: profileName,
   description: profileDescription,
+  avatar: profileAvatar,
 });
 
-const popupProfile = new PopupWithForm(
+buttonEditProfile.addEventListener("click", () => {
+  popupProfileInfo.setInputValues(userProfile.getUserInfo()); // { name, description }
+  formValidators["profile_info_form"].resetFormValidation(); // input always valid when taking from the html page
+  popupProfileInfo.open();
+});
+
+const popupProfileInfo = new PopupWithForm(
   modalWindowProfile,
   (event, { name, description }) => {
     event.preventDefault();
-    userProfile.setUserInfo(name, description);
-    popupProfile.close();
+    handleSubmit(() => {
+      return api.setUserProfileData(name, description).then(() => {
+        userProfile.setUserInfo(name, description);
+      });
+    }, popupProfileInfo);
   },
-  configuration.inputSelector
+  configuration.inputSelector,
+  submitProfileInfo
 );
-popupProfile.setEventListeners();
+popupProfileInfo.setEventListeners();
+/* END PROFILE INFO SECTION */
 
-buttonEditProfile.addEventListener("click", () => {
-  const { name, description } = userProfile.getUserInfo();
-  inputProfileName.value = name;
-  inputProfileDescription.value = description;
-  formValidators["profile_form"].resetFormValidation(); // input always valid when taking from the html page
-  popupProfile.open();
+/* PROFILE AVATAR SECTION */
+buttonUpdateAvatar.addEventListener("click", () => {
+  popupProfileAvatar.setInputValues(userProfile.getAvatar()); // { avatar } as obj, not string
+  popupProfileAvatar.open();
 });
 
-/* CARD ADD SECTION */
-const popupCardAdd = new PopupWithForm(
-  modalWindowCardAdd,
-  (event, cardInputs) => {
+const popupProfileAvatar = new PopupWithForm(
+  modalWindowProfileAvatar,
+  (event, { avatar }) => {
     event.preventDefault();
-    const cardElement = createCard(cardInputs);
-    cardSection.addItem(cardElement, "prepend"); // can still be used to add later cards, not just init ones
-    popupCardAdd.close();
-    formCardAdd.reset();
-    formValidators["card-add_form"].disableButton();
+    handleSubmit(() => {
+      return api.setUserProfileAvatar(avatar).then(() => {
+        userProfile.setAvatar(avatar);
+        popupProfileAvatar.close();
```

Comment:
`        popupProfileAvatar.close();
`

It's not needed because `handleSubmit` contains closing